bark/world/map/commons.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// Authors: Julian Bernhard, Klemens Esterle, Patrick Hart and
+// Tobias Kessler
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+#ifndef BARK_WORLD_MAP_COMMONS_HPP_
+#define BARK_WORLD_MAP_COMMONS_HPP_
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "bark/geometry/geometry.hpp"
+#include "bark/world/world.hpp"
+
+namespace bark {
+namespace world {
+namespace map {
+
+using bark::geometry::Point2d;
+using bark::geometry::Polygon;
+using bark::geometry::Within;
+using bark::world::ObservedWorld;
+namespace bg = boost::geometry;
+using models::dynamic::StateDefinition;
+using bark::geometry::SignedAngleDiff;
+
+/**
+ * @brief  Function that selects LaneCorridors
+ * @note   If an agent is in two LaneCorridors with its shape
+ *         the other one it selected if the front points of the
+ *         agent are within the other lane corridor.
+ *         This models an entering of the other lane corridor.
+ * @param  observed_world: ObservedWorld of agent
+ * @param  target_corr: fallback corridor in case calculation fails
+ * @retval LaneCorridorPtr: the computed lane corridor
+ */
+inline LaneCorridorPtr ChooseLaneCorridorBasedOnVehicleState(
```

Comment:
I like it! Do we have a unit test for it?

Reply:
I think we can discuss how we could unit test this next week.
Then it should be possible to still implement it until Thursday!

Reply:
No